oxidation/libparsec/crates/tests_fixtures/src/testbed.rs

```diff
@@ -51,27 +53,56 @@ macro_rules! run_in_testbed {
             // leak until the 10mn auto-garbage collection otherwise.
             scope.stop().await;
         }
-    };
-}
-#[macro_export]
-macro_rules! run_in_testbed_with_server {
-    ($template:literal, $env:ident => $cb:expr) => {
-        run_in_testbed!($template, true, $env => $cb)
-    };
+    }
+
+    pub async fn run<F, Fut>(template: &str, cb: F)
+    where
+        F: FnOnce(Arc<TestbedEnv>) -> Fut,
+        Fut: Future<Output = ()>,
+    {
+        // Here we will skip if TESTBED_SERVER env is not set !
+        if let Some(scope) = Self::start(template, Run::WithoutServer).await {
+            // TODO: handle async panic
+            cb(scope.env.clone()).await;
+            // Note in case `cb` panics we won't be able to cleanup the testbed env :'(
+            // This is "ok enough" considering:
+            // - Using `std::panic::{catch_unwind, resume_unwind}` is cumbersome with async closure
+            // - Panic only occurs when a test fails, so at most once per run (unless
+            //   `--keep-going` is used, but that's a corner case ^^).
+            // Hence leak should be small: no leak if the testbed server has been started by us,
+            // leak until the 10mn auto-garbage collection otherwise.
+            scope.stop().await;
+        }
+    }
 }
-pub use run_in_testbed;
-pub use run_in_testbed_with_server;
 
 /// Testbed server is lazily started when it is first needed, and then shared
 /// across all tests.
 /// Alternatively, `TESTBED_SERVER` environ variable can be passed to use an
 /// external server.
-fn ensure_testbed_server_is_started() -> BackendAddr {
-    if let Ok(raw) = std::env::var("TESTBED_SERVER") {
-        let addr = BackendAddr::from_any(&raw)
+fn ensure_testbed_server_is_started() -> Option<BackendAddr> {
+    // TESTBED_SERVER should have 3 possible values:
+    // - not set, if so the test is skipped (don't forget to print something about it !)
+    // - `AUTOSTART` or `1` or just defined to empty: auto start the serve
+    // - parsec://<domain>:<port>[?no_ssl=true],
+    //   <domain>:<port>,
+    //   http://<domain>:<port> or
+    //   https://<domain>:<port> consider the server is already started
+    let testbed_server = std::env::var("TESTBED_SERVER");
+
+    if let Err(e) = testbed_server {
+        // That means the env variable isn't set
+        println!("{e}, so the test is skipped !");
```

Comment:
```suggestion
        eprintln!("`TESTBED_SERVER` env var isn't set, skipping the test ! ({e})");
```

Reply:
Why `stderr` ? This is not an error

Reply:
the value of `e` can change, It by doing the review I don't known it's value beforehand